open_academy/models/session.py

```diff
@@ -1,13 +1,48 @@
-from odoo import models, fields
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
 
 
 class Session(models.Model):
     _name = 'session'
     _description = 'Session'
     course_id = fields.Many2one('course', required=True)
-    instructor_id = fields.Many2one('res.partner')
+    instructor_id = fields.Many2one('res.partner', domain=(
+        "['|',"
+        "('is_instructor', '=', 'True'),"
+        "('category_id.name', 'like', 'Teacher'),]"),
+    )
     attendee_ids = fields.Many2many('res.partner')
     name = fields.Char(required=True)
     start_date = fields.Date()
     duration = fields.Float()
-    number_seat = fields.Integer()
+    number_seat = fields.Integer(default=1, required=True)
+    percentage_taken_seat = fields.Float(compute='_calculate_percentage_taken_seat')
+
+    @api.depends('attendee_ids', 'number_seat')
+    def _calculate_percentage_taken_seat(self):
+        for record in self:
+            record.percentage_taken_seat = (len(record.attendee_ids) / record.number_seat) * 100 if record.number_seat > 0 else 0
+
+    @api.onchange('taken_seat', 'number_seat')
+    def _onchange_percentage_taken_seat(self):
+        if self.number_seat <= 0 or self.taken_seat < 0:
+            return {
+                'warning':{
+                    'title': 'Values not allowed!',
+                    'message': ('Number seat can not be negative or zero or Taken seat can not be negative'
+                    ),
+                }
+            }
+        elif self.taken_seat > self.number_seat:
+            return {
+                'warning': {
+                    'title': 'Taken seat value not allowed!',
+                    'message': 'Taken seat can not be greater than Number seat',
+                }
+            }
```

Comment:
`taken_seat` this field does not exist anymore. Please test the code before tag me.